### PR TITLE
Ensure unique names for apps deployed in multiple clusters

### PR DESCRIPTION
**Pull request by larsks**
Use the nameSuffix transformer [1] to ensure unique names for apps that are deployed in multiple clusters.

This commit also introduces a `lib/` directory to contain common application manifests that can then be included and patched in clusters that require that application.

[1]: https://github.com/kubernetes-sigs/kustomize/blob/master/examples/transformerconfigs/README.md#prefixsuffix-transformer

### CODE REVIEW

**Review by computate — APPROVED** [review]

**Comment by larsks** [comment]
This has been rebased on (and includes) #20 and should not merge until #20 has merged.

**Review by naved001 — APPROVED** [review]

**Pull request merged**
